src/java.desktop/share/classes/com/sun/beans/finder/MethodFinder.java

```diff
@@ -77,8 +77,7 @@ public static Method findMethod(Class<?> type, String name, Class<?>...args) thr
         Signature signature = new Signature(type, name, args);
 
         try {
-            Method method = CACHE.get(signature);
-            return (method == null) ? method : CACHE.create(signature);
+            return CACHE.get(signature);
```

Comment:
Is it possible to create a test for this patch?

Reply:
We have a slowdown [JDK-8350573](https://bugs.openjdk.org/browse/JDK-8350573) of the `java/beans/XMLDecoder/8028054/TestMethodFinder.java` test

The possible test would be to set a timeout for it, but is depends on the hardware:

> On my test machine the test runtime goes from ~ 40s (hash of the change before [JDK-8344891](https://bugs.openjdk.org/browse/JDK-8344891)) to ~ 1:40 ([JDK-8344891](https://bugs.openjdk.org/browse/JDK-8344891)) .

> Locally on my M1 Macbook pro this test runs in ~4 seconds before and in ~28 seconds after [JDK-8344891](https://bugs.openjdk.org/browse/JDK-8344891) change.

So I believe it would be better to track anomalies by test execution time history in the CI.